stake_vote/src/utils.rs

```diff
@@ -21,6 +27,52 @@ pub async fn get_current_epoch(data_cache: &DataCache) -> LiteRpcEpoch {
     data_cache.get_current_epoch(commitment).await
 }
 
+//Read save epoch vote stake to bootstrap current leader shedule and get_vote_account.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+struct StringSavedStake {
+    epoch: u64,
+    stake_vote_map: HashMap<String, (u64, Arc<StoredVote>)>,
+}
+
+#[allow(clippy::type_complexity)]
+pub fn read_schedule_vote_stakes(
+    file_path: &str,
+) -> anyhow::Result<(u64, HashMap<Pubkey, (u64, Arc<StoredVote>)>)> {
+    let content = std::fs::read_to_string(file_path)?;
+    let stakes_str: StringSavedStake = serde_json::from_str(&content)?;
+    //convert to EpochStake because json hashmap parser can only have String key.
+    let ret_stakes = stakes_str
+        .stake_vote_map
+        .into_iter()
+        .map(|(pk, st)| (Pubkey::from_str(&pk).unwrap(), (st.0, st.1)))
+        .collect();
+    Ok((stakes_str.epoch, ret_stakes))
+}
+
+pub fn save_schedule_vote_stakes(
+    base_file_path: &str,
+    stake_vote_map: &HashMap<Pubkey, (u64, Arc<StoredVote>)>,
+    epoch: u64,
+) -> anyhow::Result<()> {
+    //save new schedule for restart.
+    //need to convert hahsmap key to String because json aloow only string
+    //key for dictionnary.
+    //it's better to use json because the file can use to very some stake by hand.
+    //in the end it will be removed with the bootstrap process.
+    let save_stakes = StringSavedStake {
+        epoch,
+        stake_vote_map: stake_vote_map
+            .iter()
+            .map(|(pk, st)| (pk.to_string(), (st.0, Arc::clone(&st.1))))
+            .collect(),
+    };
+    let serialized_stakes = serde_json::to_string(&save_stakes).unwrap();
+    let mut file = File::create(base_file_path).unwrap();
+    file.write_all(serialized_stakes.as_bytes()).unwrap();
+    file.flush().unwrap();
+    Ok(())
+}
+
 //Takable struct code
 pub trait TakableContent<T>: Default {
```

Comment:
This seems to be de coupled from this crate. Move this structure to core/src/structures.

Reply:
It's a specific pattern implementation use only by vote and stake. I prefer to let it there and move it if it's needed in another crate. In fact, it's going to have a lot of change and to be very specific to the stake and vote data management.